### ISSUE TITLE
[TECHDEBT] [Persistence] Fix Persistence Module TODOs - Golang Test Cleanup & Teardown

### ISSUE THREAD

**Issue by jessicadaugherty** [pokt-network/pocket#208]
# Objective
A general cleanup issue is needed to tackle TODO's and ensure the persistence module is usable/readable by refactoring golang-specific Golang test cleanup & teardown as part of https://github.com/pokt-network/pocket/issues/172 and https://github.com/pokt-network/pocket/issues/149. 

# Origin Document
Should follow issue-#128, issue-#105, issue-#147 and issue-#148 the persistence module is messier and more difficult to understand than the developers would want for organic external contribution.

# Goals / Deliverables
- [ ] Implement golang-specific Golang test cleanup & teardown.
For example, the usage of:
```go
func CleanupTest() {
	PostgresDB.Tx.Rollback(context.TODO())
	PersistenceModule.Stop()
}
```
Can be refactored to something like this:

```go
func NewTestingUtilityContext(t *testing.T, height int64) utility.UtilityContext {   
   ... 
   t.Cleanup(func() {
        ctx.Context.Release() // TODO (team) need a golang specific solution for teardown
        PostgresDB.Tx.Rollback(context.TODO())
        PersistenceModule.Stop()
   })
  ...
  return &UtilityContext{...
```

## General issue checklist
- [ ] Update the appropriate CHANGELOG
- [ ] Update the README
- [ ] If applicable, update the source code tree explanation
- [ ] If applicable, add or update a state, sequence or flowchart diagram using [mermaid](https://mermaid-js.github.io/mermaid/)
- [ ] Update any relevant global documentation & references
- [ ] Document small issues / TODOs along the way
---

Creator: @andrewnguyen22
Co-creator: @Olshansk

**Comment by Olshansk** [comment]
@Jasonyou1995 I assigned this task to you since you've been digging into the persistence module. No need to tackle it right away (we can figure that part out) but the main goal here is to move our proprietary testing setup & teardown to the standard libraries use in Go.

This is a good article as a starting point to learn more: https://medium.com/nerd-for-tech/setup-and-teardown-unit-test-in-go-bd6fa1b785cd

The tl;dr is to remove all instance of `test_artifacts.CleanupTest(ctx)` which are spread across the `utility`, `persistence` and `runtime` modules in favor of using Go's `t.Cleanup`

**Comment by Olshansk** [comment]
Resolved as part of #550.
